src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -81,22 +81,47 @@ end
 
 
 # allow passing named tuples of BCs constructed in an arbitrary order
-digest_boundary_conditions(boundary_conditions) = boundary_conditions
+digest_boundary_conditions(boundary_conditions, cache) = boundary_conditions
 
-function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}) where {Keys, ValueTypes<:NTuple{2,Any}} # 1D
+function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}, cache) where {Keys, ValueTypes<:NTuple{2,Any}} # 1D
   @unpack x_neg, x_pos = boundary_conditions
   (; x_neg, x_pos)
 end
-function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}) where {Keys, ValueTypes<:NTuple{4,Any}} # 2D
+function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}, cache) where {Keys, ValueTypes<:NTuple{4,Any}} # 2D
   @unpack x_neg, x_pos, y_neg, y_pos = boundary_conditions
   (; x_neg, x_pos, y_neg, y_pos)
 end
-function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}) where {Keys, ValueTypes<:NTuple{6,Any}} # 3D
+function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}, cache) where {Keys, ValueTypes<:NTuple{6,Any}} # 3D
   @unpack x_neg, x_pos, y_neg, y_pos, z_neg, z_pos = boundary_conditions
   (; x_neg, x_pos, y_neg, y_pos, z_neg, z_pos)
 end
 
-function digest_boundary_conditions(boundary_conditions::AbstractArray)
+# peel apart the boundary condition dictionary for the UnstructuredQuadMesh solver and sort the
+# boundary types and their associated boundary indices into tuples for use later
+function digest_boundary_conditions(boundary_conditions::Dict, cache)
+  # extract the unique boundary function routines from the dictionary
+  boundary_types = Tuple(unique(collect(values(boundary_conditions))))
+
+  # pull and sort the indexing for each boundary type
+  boundary_indices = Vector{Any}(nothing, length(boundary_types))
+  for j in 1:length(boundary_types)
+    indices_for_current_type = Int[]
+    for (test_name, test_condition) in boundary_conditions
+      temp_indices = findall(x->x===test_name, cache.boundaries.name)
+      if test_condition === boundary_types[j]
+        indices_for_current_type = vcat(indices_for_current_type, temp_indices)
+      end
+    end
+    boundary_indices[j] = sort(indices_for_current_type)
```

Comment:
```suggestion
    boundary_indices[j] = sort!(indices_for_current_type)
```
Minor performance optimization